### ISSUE TITLE
[Question] Proper FastRTPS version for SOSS-DDS

### ISSUE THREAD

**Issue by kteriusghub** [eProsima/SOSS-DDS#46]
Hello, eProsima members!
I am trying to use SOSS-DDS for communication between two ROS2 nodes on TCP tunneling.
And, I have the same issue as #20 and #41 ,  and eProsima engineers gave the kind guides to them.
Anyway, the point of the guide is the proper version of FastRTPS for Crystal and Dashing.
My FastRTPS is v1.8.2 of the latest Dashing, however, for well-working SOSS-DDS, richiware and lemunozm mentioned that v1.7.2 is proper for Crystal and v1.8.3 for Dashing. And, additionally, richiware said that v1.8.2 has the incompatibility issue. (BTW, in #41 , gek40 said that 1.9.x showed the good result for SOSS-DDS on Dashing.)
I should consider the proper FastRTPS version on Dashing, because I should manage it for all ROS2 machines in the office and compatibility to other 3rd party company.
Is there any side effects to apply v1.8.3 to Dashing? With 1.8.3, no compatibility issue to another remote ROS2 machines with v1.8.2?
How about 1.9.x about the same case of Dashing environment?

**Comment by richiware** [comment]
> Is there any side effects to apply v1.8.3 to Dashing?

If you compile ROS2 with Fast-RTPS 1.8.3 there should be no problems.

> With 1.8.3, no compatibility issue to other remote ROS2 machines with v1.8.2?

Probably there will be no communication between 1.8.3 and 1.8.2 due to the issue introduced on 
1.8.2. I recommend you to use 1.8.3 in all machines.

>How about 1.9.x about the same case of Dashing environment?

No problem either. There should be no problems using this version for Dashing. 1.9.x has different ABI that 1.8.x, but this is only a problem when you have precompiled ROS2 libraries. If you compile all ROS2 Dashing with 1.9.x, you should not have any problem.

**Comment by kteriusghub** [comment]
@richiware  
I applied both of 1.8.3 and 1.9.4 to my dashing environment and SOSS-DDS.
And, the test result was the same as gek40's  eProsima/FastDDS-SH#41 .
He mentioned as the following and he closed the issue :
  "Fast-RTPS v1.8.3 - compile is fine. But the problem remains the same.
  "Fast-RTPS v1.9.4 - compile is fine. 
         Communication ROS2 <-> Fast-RTPS node works in both direction"

For SOSS-DDS on ROS2 dashing, I think that 1.9.4 is the only way.
Just, my colleagues use the debian packages of ROS2 dashing instead of source building of ROS2.

I close the issue of question, and thanks so much to help me. 
Anyway, with 1.9.4, I faced the new issue and the issue looks similar as eProsima/FastDDS-SH#16 .
If I can't solve it, I will register a new issue to get the help from you.
